modules/tensor_mechanics/include/userobjects/RankTwoOnLine.h

```diff
@@ -0,0 +1,56 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef RANKTWOONLINE_H
+#define RANKTWOONLINE_H
+
+#include "ElementUserObject.h"
+#include "RankTwoAux.h"
+
+class RankTwoOnLine;
+class RankTwoTensor;
+
+template<>
+InputParameters validParams<RankTwoOnLine>();
+
```

Comment:
Add in the Doxygen comments in here to explain what this class does--touch base with Ben if need be